### ISSUE TITLE
display failure reason on Runs page

### ISSUE THREAD

**Issue by peterbaccaro** [dagster-io/dagster#6159]
## Use Case

On the Runs page, you can apply Filter of ‘status:FAILURE’ to see the failures, however, the reason for each failure is not displayed.  In order to see the failure reason, you need to drill into the run instance and determine the error by looking at the instance event log.

## Ideas of Implementation

Displaying the error reason on the Runs page will enable quick identification of the types of errors that have occurred which would also be useful when deciding if a failed execution should be re-executed.

## Additional Info

---
#### Message from the maintainers:
Excited about this feature? Give it a :thumbsup:. We factor engagement into prioritization.

**Comment by dpeng817** [comment]
For reference, this is what the page currently looks like when filtering on failure: 
<img width="1436" alt="Screen Shot 2022-01-10 at 2 40 54 PM" src="https://user-images.githubusercontent.com/17576880/148850379-ced1e3a8-482c-4d68-a999-1aa601ca1cea.png">
It's possible that we could add a tooltip, or some sort of indicator of the error type on the row. 
@hellendag thoughts?

**Comment by hellendag** [comment]
@dpeng817 Can errors generally be distilled or simplified into common messages that would fit in a tooltip? I don't think we would want to retrieve stack traces from logs and display them here.

One note: if you hover on the "Failure" tag, it will give you a link to the failed steps, which should jump directly to them in the event log. Is this helpful, or not really enough for this case?

**Comment by dpeng817** [comment]
@hellendag I was thinking more the error type than the message, but arguably that's not super informative. In addition, while it might make sense to see error reasoning when you have a small number of runs, in a prod setting this can quickly become information overload. @peterbaccaroshell does the failed steps link help in any way? Perhaps this front-end functionality is not well-enough documented.

**Comment by peterbaccaroshell** [comment]
@dpeng817  No, that doesn't help.  The purpose would be to know what error occurred without navigating to the run instance to look through the trace log.  There may be a situation where you want to re-execute Runs that have failed with a specific error so having to navigate to the Runs page to determine if the Run has failed for this reason is painful. Another scenario is when you want to know how many of each type of error has occurred (viewing the trace log for each failure would be very time consuming).  
The error type alone would not be perfect but would still be better than having no error information displayed at all.  The error tag could be created when the error occurs to avoid the UI having to retrieve it from the trace log if that was an issue.

**Comment by sryza** [comment]
> I don't think we would want to retrieve stack traces from logs and display them here.

@hellendag - is this mainly for performance reasons?

**Comment by peterbaccaroshell** [comment]
Are there any plans to implement something for this?  I was thinking that the Runs page could display the error type.  Ideally this would be something that could be filtered on also.
